Sources/Core/Renderer/GUI/GUI.swift

```diff
@@ -306,54 +312,66 @@ struct GUI {
       gpuTimeString = String(format: ", %.02fms gpu", gpuTime)
     }
     let fpsString = String(format: "%.00f", renderStatistics.averageFPS)
-    list.add("\(fpsString) fps\(theoreticalFPSString) (\(cpuTimeString)ms cpu\(gpuTimeString))")
+    leftList.add("\(fpsString) fps\(theoreticalFPSString) (\(cpuTimeString)ms cpu\(gpuTimeString))")
 
     // Dimension
-    list.add("Dimension: \(client.game.world.dimension.identifier)")
-    list.add(spacer: 6)
+    leftList.add("Dimension: \(client.game.world.dimension.identifier)")
+    leftList.add(spacer: 6)
 
     // Position
     let x = String(format: "%.02f", position.x)
     let y = String(format: "%.02f", position.y)
     let z = String(format: "%.02f", position.z)
-    list.add("XYZ: \(x) / \(y) / \(z)")
+    leftList.add("XYZ: \(x) / \(y) / \(z)")
 
     // Block under feet
-    list.add("Block: \(blockPosition.x) \(blockPosition.y) \(blockPosition.z)")
+    leftList.add("Block: \(blockPosition.x) \(blockPosition.y) \(blockPosition.z)")
 
     // Chunk section and relative position
     let relativePosition = blockPosition.relativeToChunkSection
     let relativePositionString = "\(relativePosition.x) \(relativePosition.y) \(relativePosition.z)"
     let chunkSectionString = "\(chunkSectionPosition.sectionX) \(chunkSectionPosition.sectionY) \(chunkSectionPosition.sectionZ)"
-    list.add("Chunk: \(relativePositionString) in \(chunkSectionString)")
+    leftList.add("Chunk: \(relativePositionString) in \(chunkSectionString)")
 
     // Heading and rotation
     let yawString = String(format: "%.01f", yaw)
     let pitchString = String(format: "%.01f", pitch)
-    list.add("Facing: \(heading) (Towards \(heading.isPositive ? "positive" : "negative") \(heading.axis)) (\(yawString) / \(pitchString))")
+    leftList.add("Facing: \(heading) (Towards \(heading.isPositive ? "positive" : "negative") \(heading.axis)) (\(yawString) / \(pitchString))")
 
     // Lighting (at foot level)
     var lightPosition = blockPosition
     lightPosition.y += 1
     let skyLightLevel = client.game.world.getSkyLightLevel(at: lightPosition)
     let blockLightLevel = client.game.world.getBlockLightLevel(at: lightPosition)
-    list.add("Light: \(skyLightLevel) sky, \(blockLightLevel) block")
+    leftList.add("Light: \(skyLightLevel) sky, \(blockLightLevel) block")
 
     // Biome
     let biome = client.game.world.getBiome(at: blockPosition)
-    list.add("Biome: \(biome?.identifier.description ?? "not loaded")")
+    leftList.add("Biome: \(biome?.identifier.description ?? "not loaded")")
 
     // Gamemode
-    list.add("Gamemode: \(gamemode.string)")
+    leftList.add("Gamemode: \(gamemode.string)")
+
+    var rightList = GUIList(rowHeight: 9, renderRowBackground: true, rightAlign: true)
+
+    var cpuName = HWInfo.CPU.name()
```

Comment:
It'd be best to make all four of these variables just static properties of `GUI` so that they only load once. Most of them are pretty light to load every frame, but `GPUDetection` creates the default system metal device every time which I've heard isn't a great idea to do every frame (better safe than sorry). You should be able to just cut and paste these up to line 17 and then change `var` to `static let`. It should probably be `let` anyway afaict.